test/IECoreScene/MeshAlgoDistributePointsTest.py

```diff
@@ -176,6 +178,36 @@ def testVertexUVs( self ) :
 
 		self.assertEqual( p, p2 )
 
+	def testCancel( self ) :
+		# Initializing the points distribution is slow and not cancellable
+		# Pre-initialize it so it doesn't mess with our timing
+		IECore.PointDistribution.defaultInstance()
+
+		canceller = IECore.Canceller()
+		cancelled = [False]
+
+		def backgroundRun():
+			m = IECore.Reader.create( "test/IECore/data/cobFiles/pCubeShape1.cob" ).read()
+			try:
+				p = IECoreScene.MeshAlgo.distributePoints( mesh = m, density = 10000000, offset = imath.V2f( 0.0001, 0.0001 ), canceller = canceller )
+			except IECore.Cancelled:
+				cancelled[0] = True
+
+		thread = threading.Thread(target=backgroundRun, args=())
+
+		startTime = time.time()
+		thread.start()
+
+		time.sleep( 0.1 )
```

Comment:
What is the sleep for? In some of the Gaffer tests it is an attempt to avoid the built-in cancellation at the process level so we can test the cancellation within a compute, but that doesn't apply here.

Reply:
As discussed this morning, I felt like it made sense to test a cancel once it started running, and we were catching it in the middle of processing, rather than testing a cancel that occurs immediately ( which might be testing the prep code more than the body of the function ).